clouddriver-alicloud/src/main/java/com/netflix/spinnaker/clouddriver/alicloud/cache/Keys.java

```diff
@@ -0,0 +1,291 @@
+      String loadBalancerName, String account, String region, String vpcId) {
+    String key =
+        ID
+            + SEPARATOR
+            + Namespace.LOAD_BALANCERS
+            + SEPARATOR
+            + account
+            + SEPARATOR
+            + region
+            + SEPARATOR
+            + loadBalancerName;
+    if (!StringUtils.isEmpty(vpcId)) {
+      key = key + SEPARATOR + vpcId;
+    }
+    return key;
+  }
+
+  public static String getSubnetKey(String vSwitchId, String region, String account) {
+    String key =
+        ID
+            + SEPARATOR
+            + Namespace.SUBNETS
+            + SEPARATOR
+            + account
+            + SEPARATOR
+            + region
+            + SEPARATOR
+            + vSwitchId;
+    return key;
+  }
+
+  public static String getImageKey(String imageId, String account, String region) {
+    String key =
+        ID
+            + SEPARATOR
+            + com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace.IMAGES
+            + SEPARATOR
+            + account
+            + SEPARATOR
+            + region
+            + SEPARATOR
+            + imageId;
+    return key;
+  }
+
+  public static String getNamedImageKey(String account, String imageName) {
+    String key =
+        ID
+            + SEPARATOR
+            + com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace.NAMED_IMAGES
+            + SEPARATOR
+            + account
+            + SEPARATOR
+            + imageName;
+    return key;
+  }
+
+  public static String getInstanceTypeKey(String account, String region, String zoneId) {
+    String key =
+        ID
+            + SEPARATOR
+            + Namespace.INSTANCE_TYPES
+            + SEPARATOR
+            + account
+            + SEPARATOR
+            + region
+            + SEPARATOR
+            + zoneId;
+    return key;
+  }
+
+  public static String getSecurityGroupKey(
+      String securityGroupName,
+      String securityGroupId,
+      String region,
+      String account,
+      String vpcId) {
+    String key =
+        ID
+            + SEPARATOR
+            + Namespace.SECURITY_GROUPS
+            + SEPARATOR
+            + account
+            + SEPARATOR
+            + region
+            + SEPARATOR
+            + securityGroupName
+            + SEPARATOR
+            + securityGroupId;
+    if (!StringUtils.isEmpty(vpcId)) {
+      key = key + SEPARATOR + vpcId;
+    }
+    return key;
+  }
+
+  public static String getKeyPairKey(String keyPairName, String region, String account) {
+    String key =
+        ID
+            + SEPARATOR
+            + Namespace.ALI_CLOUD_KEY_PAIRS
+            + SEPARATOR
+            + keyPairName
+            + SEPARATOR
+            + account
+            + SEPARATOR
+            + region;
+    return key;
+  }
+
+  public static String getServerGroupKey(
+      String autoScalingGroupName, String account, String region) {
+    Names names = Names.parseName(autoScalingGroupName);
```

Comment:
You should be using `Moniker` here instead of `Names`.

Reply:
HI @robzienert Thanks for your review. How to use `Moniker`? Is there any examples?

Reply:
Sure, there's a [NamerRegistry](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/names/NamerRegistry.java) you can use to get access to the moniker for a particular account config. 

Here's an example use of it in the [`SecurityGroup`](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/SecurityGroup.java#L61) model.

cc @ezimanyi Do you have a good example of Moniker usage aside from what I've provided?

Reply:
Here's [an example](https://github.com/spinnaker/clouddriver/blob/9017ad54e09e92dfeafb2891b654977055caa791/clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/agent/KubernetesCacheDataConverter.java#L176) from the Kubernetes provider.